org.jacoco.core/src/org/jacoco/core/analysis/Analyzer.java

```diff
@@ -143,13 +141,11 @@ public void analyzeClass(final byte[] buffer, final String location)
 	 */
 	public void analyzeClass(final InputStream input, final String location)
 			throws IOException {
-		final byte[] buffer;
 		try {
-			buffer = Java9Support.readFully(input);
-		} catch (final IOException e) {
-			throw analyzerError(location, e);
+			analyzeClass(new ClassReader(input));
+		} catch (final Exception cause) {
```

Comment:
And here from IOException to Exception

Reply:
Also wondering why? and how this is related to ASM 6?

Reply:
Same comment here, I got a test with an exception (I think was an `ArrayIndexOutOfBoundsException` in this case)

Reply:
But reverting this change I got an `Failed tests:   testAnalyzeClass_BrokenStream(org.jacoco.core.analysis.AnalyzerTest): expected:<Error while analyzing BrokenStream.> but was:<null>`